statsmodels/multivariate/factor.py

```diff
@@ -0,0 +1,272 @@
+
+
+class Factor(Model):
+    """
+    Factor analysis
+
+    .. [1] Hofacker, C. (2004). Exploratory Factor Analysis, Mathematical Marketing.
+    http://www.openaccesstexts.org/pdf/Quant_Chapter_11_efa.pdf
+
+    Supported rotations:
+        'varimax', 'quartimax', 'biquartimax', 'equamax', 'oblimin',
+        'parsimax', 'parsimony', 'biquartimin', 'promax'
+
+    Parameters
+    ----------
+    endog : array-like
+        Variables in columns, observations in rows
+
+    n_factor : int
+        The number of factors to extract
+
+    method : str
+        Specify the method to extract factors.
+        'pa' - Principal axis factor analysis
+
+    """
+    def __init__(self, endog, n_factor, exog=None, method='pa', **kwargs):
+        if n_factor <= 0:
+            raise ValueError('n_factor must be larger than 0! %d < 0' %
+                             (n_factor))
+        if n_factor > endog.shape[1]:
+            raise ValueError('n_factor must be smaller or equal to the number'
+                             ' of columns of endog! %d > %d' %
+                             (n_factor, endog.shape[1]))
+        self.n_factor = n_factor
+        self.loadings = None
+        self.communality = None
+        self.eigenvals = None
+        self.method = method
+        super(Factor, self).__init__(endog, exog)
+
+    def fit(self, **kwargs):
+        """
+        Extract factors
+        """
+        if self.method == 'pa':
+            return self.fit_pa(**kwargs)
+        else:
+            raise ValueError("Unknown factor extraction approach '%s'" % self.method)
+
+    def fit_pa(self, n_max_iter=50, tolerance=1e-6, SMC=True):
+        """
+        Extract factors using the iterative principal axis method
+
+        Parameters
+        ----------
+        n_max_iter : int
+            Maximum number of iterations for communality estimation
+        tolerance : float
+            If `norm(communality - last_communality)  < tolerance`,
+            estimation stops
+        SMC : True or False
+            Whether or not to apply squared multiple correlations
+
+        -------
+
+        """
+        R = np.corrcoef(self.endog, rowvar=0)
+        self.n_comp = matrix_rank(R)
+        if self.n_factor > self.n_comp:
+            raise ValueError('n_factor must be smaller or equal to the rank'
+                             ' of endog! %d > %d' %
+                             (self.n_factor, self.n_comp))
+        if n_max_iter <= 0:
+            raise ValueError('n_max_iter must be larger than 0! %d < 0' %
+                             (n_max_iter))
+        if tolerance <= 0 or tolerance > 0.01:
+            raise ValueError('tolerance must be larger than 0 and smaller than'
+                             ' 0.01! Got %f instead' % (tolerance))
+        #  Initial communality estimation
+        if SMC:
+            c = 1 - 1 / np.diag(inv(R))
+            self.SMC = np.array(c)
+        else:
+            c = np.ones([1, len(R)])
```

Comment:
this should be 1 dimensional, AFAICS
`c = np.ones(len(R)) `
then there is no exception

unit tests ?